### ISSUE TITLE
Dependent packages should only be upgraded once

### ISSUE THREAD

**Issue by suominen** [NetBSDfr/pkgin#128]
If you install e.g. the `git` package and it then gets an upgrade, `pkgin` will upgrade the dependent packages of `git` twice: once for the meta-package and a second time for each dependent package.

It is only necessary to upgrade each package once, and that would be the desired behaviour.

<details>
<summary>Click to view an example <code>pkgin ug</code> run.</summary>

```
: revolutions:~# pkgin ug
pkg_summary.xz                                                                                  100%   56KB  55.6KB/s   00:00
calculating dependencies...done.

7 packages to refresh:
  gw-dev-2.5 bash-completion-2.11 gw-base-2.5 gh-2.20.2nb1 tig-2.5.7 rsync-3.2.7 mosh-1.3.2nb8

8 packages to upgrade:
  tcsh-6.24.07 protobuf-3.21.12 git-docs-2.39.0 git-contrib-2.39.0 git-2.39.0 curl-7.87.0 bash-5.2.15 git-base-2.39.0

7 to refresh, 8 to upgrade, 0 to install
0B to download, 294K to install

proceed ? [Y/n]
upgrading tcsh-6.24.07...
===========================================================================
The following lines can be added to /etc/shells:

        /usr/pkg/bin/tcsh

===========================================================================
upgrading protobuf-3.21.12...
refreshing gw-dev-2.5...
upgrading git-docs-2.39.0...
upgrading git-contrib-2.39.0...
upgrading git-2.39.0...
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/description to /usr/pkg/share/git-core/templates/description
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/applypatch-msg.sample to /usr/pkg/share/git-core/templates/hooks/applypatch-msg.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/commit-msg.sample to /usr/pkg/share/git-core/templates/hooks/commit-msg.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/post-update.sample to /usr/pkg/share/git-core/templates/hooks/post-update.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/pre-applypatch.sample to /usr/pkg/share/git-core/templates/hooks/pre-applypatch.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/pre-commit.sample to /usr/pkg/share/git-core/templates/hooks/pre-commit.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/pre-rebase.sample to /usr/pkg/share/git-core/templates/hooks/pre-rebase.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/prepare-commit-msg.sample to /usr/pkg/share/git-core/templates/hooks/prepare-commit-msg.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/update.sample to /usr/pkg/share/git-core/templates/hooks/update.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/info/exclude to /usr/pkg/share/git-core/templates/info/exclude
===========================================================================
$NetBSD: MESSAGE,v 1.3 2016/05/26 15:41:06 khorben Exp $

NOTE:   Pristine templates are located in:
        /usr/pkg/share/examples/git/templates.

To use the git-cvsimport repository conversion from CVS, install git-cvs.

To use the git-svn interface to Subversion, install git-svn.

===========================================================================
upgrading curl-7.87.0...
refreshing bash-completion-2.11...
===========================================================================
$NetBSD: MESSAGE,v 1.1 2009/09/25 12:29:26 wiz Exp $

To use this, source the included bash fragment in your startup
files, i.e. include something like:

  if [ -r /usr/pkg/share/bash-completion/bash_completion ]; then
    . /usr/pkg/share/bash-completion/bash_completion
  fi

in ~/.bashrc or ~/.bash_profile.

===========================================================================
upgrading bash-5.2.15...
bash-5.2.12: unregistering info file /usr/pkg/info/bash.info
===========================================================================
The following lines can be added to /etc/shells:

        /usr/pkg/bin/bash

===========================================================================
bash-5.2.15: registering info file /usr/pkg/info/bash.info
refreshing gw-base-2.5...
refreshing gh-2.20.2nb1...
upgrading git-base-2.39.0...
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/description to /usr/pkg/share/git-core/templates/description
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/applypatch-msg.sample to /usr/pkg/share/git-core/templates/hooks/applypatch-msg.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/commit-msg.sample to /usr/pkg/share/git-core/templates/hooks/commit-msg.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/post-update.sample to /usr/pkg/share/git-core/templates/hooks/post-update.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/pre-applypatch.sample to /usr/pkg/share/git-core/templates/hooks/pre-applypatch.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/pre-commit.sample to /usr/pkg/share/git-core/templates/hooks/pre-commit.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/pre-rebase.sample to /usr/pkg/share/git-core/templates/hooks/pre-rebase.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/prepare-commit-msg.sample to /usr/pkg/share/git-core/templates/hooks/prepare-commit-msg.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/hooks/update.sample to /usr/pkg/share/git-core/templates/hooks/update.sample
git-base-2.39.0: copying /usr/pkg/share/examples/git/templates/info/exclude to /usr/pkg/share/git-core/templates/info/exclude
===========================================================================
$NetBSD: MESSAGE,v 1.3 2016/05/26 15:41:06 khorben Exp $

NOTE:   Pristine templates are located in:
        /usr/pkg/share/examples/git/templates.

To use the git-cvsimport repository conversion from CVS, install git-cvs.

To use the git-svn interface to Subversion, install git-svn.

===========================================================================
refreshing tig-2.5.7...
tig-2.5.7: copying /usr/pkg/share/examples/tig/tigrc to /etc/tigrc
refreshing rsync-3.2.7...
===========================================================================
The following files should be created for rsync-3.2.7:

        /etc/rc.d/rsyncd (m=0755)
            [/usr/pkg/share/examples/rc.d/rsyncd]

===========================================================================
refreshing mosh-1.3.2nb8...
pkg_install warnings: 0, errors: 0
reading local summary...
processing local summary...
: revolutions:~#
```
</details>

**Comment by jperkin** [comment]
Sorry, didn't mean to trigger GitHub auto-close. While this is fixed, it's not yet released, so I'll do that when the next version is ready.

**Comment by jperkin** [comment]
Fixed in 23.8.0.
